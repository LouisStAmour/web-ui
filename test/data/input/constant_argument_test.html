<!doctype html>
<!--
   Copyright (c) 2012, the Dart project authors.  Please see the AUTHORS file
   for details. All rights reserved. Use of this source code is governed by a
   BSD-style license that can be found in the LICENSE file.
-->
<html lang="en">
<head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1">
  <script type="application/javascript" src="testing.js"></script>
</head>
<body>
  <element name="x-foo" extends="span">
    <template>{{myAttribute}}</template>
    <script type="application/dart">
    import 'package:web_ui/web_ui.dart';
    class XFoo extends WebComponent {
      String myAttribute = "1";
    }
    </script>
  </element>
  Expect "1 2 3 4":
  <x-foo></x-foo>
  <span is="x-foo" my-attribute="2"></span>
  <x-foo my-attribute="3"></x-foo>
  <x-foo my-attribute="{{'4'}}"></x-foo>
  <script type="application/dart">
    import 'dart:html';
    main() {
      window.postMessage('done', '*');
    }
  </script>
</body>
</html>
