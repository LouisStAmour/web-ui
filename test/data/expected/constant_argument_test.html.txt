Content-Type: text/plain
<html lang="en"><head><style>template { display: none; }</style>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1">
  <script src="../input/packages/web_ui/testing/testing.js"></script>
</head>
<body>

  Expect "1 2 3 4":
  <span is="x-foo"><shadow-root>1</shadow-root></span>
  <span is="x-foo" my-attribute="2"><shadow-root>2</shadow-root></span>
  <span is="x-foo" my-attribute="3"><shadow-root>3</shadow-root></span>
  <span is="x-foo"><shadow-root>4</shadow-root></span>



<script type="text/javascript" src="packages/shadow_dom/shadow_dom.debug.js"></script>
<script type="application/dart" src="constant_argument_test.html_bootstrap.dart"></script></body></html>
